lightsaber/processor/src/main/java/com/joom/lightsaber/processor/model/Factory.kt

```diff
@@ -17,10 +17,18 @@
 package com.joom.lightsaber.processor.model
 
 import com.joom.grip.mirrors.Type
+import com.joom.grip.mirrors.getObjectTypeByInternalName
 
 data class Factory(
   val type: Type.Object,
   val implementationType: Type.Object,
   val dependency: Dependency,
   val provisionPoints: List<FactoryProvisionPoint>
-)
+) {
+
+  companion object {
+    fun computeImplementationType(type: Type.Object): Type.Object {
+      return getObjectTypeByInternalName("${type.internalName}\$Lightsaber\$Factory")
```

Comment:
Just wondering... is this code will still work after obfuscation?

Reply:
Proguard obfuscation runs after classes are transformed